### ISSUE TITLE
LineChart animateOnDataChange crashing when data length is different from previous data

### ISSUE THREAD

**Issue by aik117** [Abhinandan-Kushwaha/react-native-gifted-charts#791]
Hi there, thanks for the great library. I am facing an issue when changing my data set on runtime based on states. I have an array which contains data (array of objects) for different purposes like hour, day, month and year and i have divided my data into 12 lengths for example the hour contains 12 items each with 5 min difference like 5 min, 10 min and all the way to 60 min. only the month array is of 10 items because it is divided into days like 3 days, 6 days all the way to 30 days. Below is my data set
```
export const giftedLineData = [
	{ 
		id: 1,
		label: "hour", 
		values: [
			{ label: "5M", value: 0 }, 
			{ label: "10M", value: 0 }, 
			{ label: "15M", value: 0 }, 
			{ label: "20M", value: 0 }, 
			{ label: "25M", value: 0 }, 
			{ label: "30M", value: 0 }, 
			{ label: "35M", value: 0 }, 
			{ label: "40M", value: 0 }, 
			{ label: "45M", value: 0 }, 
			{ label: "50M", value: 0 }, 
			{ label: "55M", value: 0 }, 
			{ label: "60M", value: 0 }
		],
		//selected: false
	},
	{ 
		id: 2,
		label: "day", 
		values: [
			{ label: "2H", value: 0 }, 
			{ label: "4H", value: 0 }, 
			{ label: "6H", value: 0 }, 
			{ label: "8H", value: 0 }, 
			{ label: "10H", value: 0 }, 
			{ label: "12H", value: 0 }, 
			{ label: "14H", value: 0 }, 
			{ label: "16H", value: 0 }, 
			{ label: "18H", value: 0 }, 
			{ label: "20H", value: 0 }, 
			{ label: "22H", value: 0 }, 
			{ label: "24H", value: 0 }, 
		],
		//selected: false
	},
	{ 
		id: 3,
		label: "month", 
		values: [
			{ label: "3D", value: 0 }, 
			{ label: "6D", value: 0 }, 
			{ label: "9D", value: 0 }, 
			{ label: "12D", value: 0 }, 
			{ label: "15D", value: 0 }, 
			{ label: "18D", value: 0 }, 
			{ label: "21D", value: 0 }, 
			{ label: "24D", value: 0 }, 
			{ label: "27D", value: 0 }, 
			{ label: "30D", value: 0 },
			// { label: "33D", value: 0 },
			// { label: "36D", value: 0 }
		],
		//selected: false
	},
	{ 
		id: 4,
		label: "year", 
		values: [
			{ label: "1MO", value: 0 }, 
			{ label: "2MO", value: 0 }, 
			{ label: "3MO", value: 0 }, 
			{ label: "4MO", value: 0 }, 
			{ label: "5MO", value: 0 }, 
			{ label: "6MO", value: 0 }, 
			{ label: "7MO", value: 0 }, 
			{ label: "8MO", value: 0 }, 
			{ label: "9MO", value: 0 }, 
			{ label: "10MO", value: 0 }, 
			{ label: "11MO", value: 0 }, 
			{ label: "12MO", value: 0 }
		],
		//selected: false
	}
];
```
and this is my LineChart component
```
    <LineChart
		isAnimated
		animateOnDataChange
		animationDuration={1000}
                onDataChangeAnimationDuration={300}
		width={400}
		data={lineChartData}
		//initialSpacing={0}
		spacing={30}
		//textColor1="black"
		//textShiftY={-8}
		//textShiftX={-10}
		textFontSize={13}
		thickness={5}
		hideRules
		//hideYAxisText
		yAxisColor="#0BA5A4"
		//showVerticalLines
		//verticalLinesColor="rgba(14,164,164,0.5)"
		xAxisColor="#0BA5A4"
		color="#0BA5A4"
		//showValuesAsDataPointsText
	/>
```
Below is a video that shows the issue i am facing

https://github.com/user-attachments/assets/43777140-4dd3-47c6-a18d-62af4e07a38f

When i click on the month and set its data the app crashes with the error **"All elements of output range should have the same number of components"**. However i have noticed that if i comment out the animateOnDataChange prop it's working fine. Any help would be appreciated.

**Comment by faljabi** [comment]
I am facing the same issue. I am setting animateOnDataChange to false for now but It would be nice to animate on complete different data change.

**Comment by Abhinandan-Kushwaha** [comment]
Hi @aik117 @faljabi Thanks for reporting this issue.
It has been fixed in versions `1.4.30` onwards.

`animateOnDataChange` now works evevn if the lehgth of data array is changed. Earlier it worked for data changes with same length and threw error on changing the length of data array.

Please use the latest version of the library.
